pkg/scheduler/framework/plugins/helper/normalize_score_test.go

```diff
@@ -31,35 +31,103 @@ func TestDefaultNormalizeScore(t *testing.T) {
 		expectedScores []int64
 	}{
 		{
-			scores:         []int64{1, 2, 3, 4},
-			expectedScores: []int64{25, 50, 75, 100},
+			scores:         []int64{1, 2, 3, 4, 5},
+			expectedScores: []int64{0, 25, 50, 75, 100},
```

Comment:
This is changing the behavior

Reply:
> This is changing the behavior

Yes, the calculated scores have changed, but their relative relationships have not, and it obeys the rule defined in the comment `normalize the scores to [0, maxPriority]`.

> I mean, we should add a new function, rather than replacing the logic for the existing one, as the semantics are slightly different.

If we do it in this way, any suggested function names? What do you think of `GenericNormalizeScore` ?

Reply:
Maybe add some comments explaining we only accept positive scores in `DefaultNormalizeScore` is enough. I'm worried that months later, this will be removed as a cleanup for unused by core packages.🤔 Out-of-tree plugins can have there own NormalizeScore functions.

Reply:
maybe `NonNegativeNormalizeScore` and `MinMaxNormalizeScore`? Since none is called default/generic, we wouldn't delete them. The comment on the former needs to be improved regardless.

Reply:
I found a better way to stay compatible with the previous behavior without changing it. It can also handle negative scores.